pkg/bff/members.go

```diff
@@ -1,39 +1,134 @@
 package bff
 
 import (
+	"context"
+	"crypto/tls"
+	"fmt"
 	"net/http"
 
 	"github.com/gin-gonic/gin"
+	"github.com/rs/zerolog/log"
+	"github.com/trisacrypto/directory/pkg/bff/api/v1"
+	"github.com/trisacrypto/directory/pkg/bff/config"
+	members "github.com/trisacrypto/directory/pkg/gds/members/v1alpha1"
+	gds "github.com/trisacrypto/trisa/pkg/trisa/gds/api/v1beta1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/protobuf/proto"
 )
 
+// ConnectMembers creates a gRPC client to the TRISA Members Service specified in the
+// configuration. This method is used to connect to both the TestNet and the MainNet and
+// to connect to mock GDS services in testing using buffconn.
+func ConnectMembers(conf config.DirectoryConfig) (_ members.TRISAMembersClient, err error) {
+	// Create the Dial options with required credentials
+	var opts []grpc.DialOption
+	if conf.Insecure {
+		opts = append(opts, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	} else {
+		opts = append(opts, grpc.WithTransportCredentials(credentials.NewTLS(&tls.Config{})))
+	}
```

Comment:
The Members server should be mTLS authenticated, so I don't think the only reason to connect insecure is to connect to a bufconn; but I've preferred to do that recently by adding dial opts to the connect members function so that in the test path we simply pass in our own options. The connect function checks if the passed in dial opts are empty, and if so, adds the mTLS connection from the configuration.